### ISSUE TITLE
Release versions are going backwards?

### ISSUE THREAD

**Issue by veqryn** [DataDog/docker-dd-agent#170]
I'm using your 11.3.585 release, and the -alpine docker container for it.

Since it came out, you've released 11.0.590 through 11.0.5110

In addition, in docker hub, the "latest" container matches the 11.0.5110 release.

Why are your releases going backwards?

Is there something wrong with 11.3?

**Comment by hkaj** [comment]
Hi @veqryn 
The version scheme is X.Y.Z with X being the major container version, Y the minor container version and Z the agent version. Z is the part you should pay attention to the most. The most recent one now is 11.0.5112 for example. But you're right, resetting from 11.3 to 11.0 was a mistake on our part and it shouldn't happen again.

**Comment by olafnorge** [comment]
Hi @hkaj 
why haven't you fixed the mistake? I do automated version compare to figure out if my images inherit from the most recent upstream images. The mistakenly added or the mistakenly resetted images do break these automated tests.

I get always told that there are more recent images available even though these are quite old. The code I am using to compare the versions look like:

```shell
printf '%s\n' \
    "datadog/docker-dd-agent:11.1.5101-alpine" \
    "datadog/docker-dd-agent:11.2.584-alpine" \
    "datadog/docker-dd-agent:11.0.5141-alpine" \
    "datadog/docker-dd-agent:11.2.585-alpine" \
    "datadog/docker-dd-agent:11.4.5141-alpine" \
    "datadog/docker-dd-agent:11.2.583-alpine" \
    "datadog/docker-dd-agent:11.3.585-alpine" \
    | sort -Vu
```

Which gives you a well sorted output:

```shell
datadog/docker-dd-agent:11.0.5141-alpine
datadog/docker-dd-agent:11.1.5101-alpine
datadog/docker-dd-agent:11.2.583-alpine
datadog/docker-dd-agent:11.2.584-alpine
datadog/docker-dd-agent:11.2.585-alpine
datadog/docker-dd-agent:11.3.585-alpine
datadog/docker-dd-agent:11.4.5141-alpine # as an example what I would expect
```

Do you know a solution to this or have an advise how to deal with it? I would recommend retagging the newer images so that they are recognized to be newer. Or simply remove the 11.x images that are greater than the 11.0.x releases.

**Comment by olafnorge** [comment]
Hi @hkaj,

I even don't get why you use such a strange versioning scheme. Why not using the agent's version as the tag for the docker image as well?

I can't find any reasonable use case why there should be one agent version that lives in several different docker image tags. I would expect that you re-use a image tag whenever you upgrade software in it but the agent version does not change. If I take a look at other players like MySQL, PostgreSQL, Java, etc they do it the same.

As far as I understand you use docker to ship the agent in an encapsulated environment nothing more nothing less.

**Comment by hkaj** [comment]
Hey @olafnorge 
Thanks for the feedback. You're right, mistakes were made in tagging some previous versions. We'll fix this in the next major (12.0.xxx) starting from which we won't downgrade minor numbers and the versioning scheme should come back to sanity.

FYI the scheme is `major.minor.agentVersion`, `major` and `minor` applying to the docker image version. We need this so we can update the docker image without updating the agent itself. As you can see in the history it happens often enough to warrant this feature.

We have a few important changes that haven't been released yet except in `latest`. We'll cut a major soon.
